tests/CppUTestExt/IEEE754PluginTest_c.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2015, Michael Feathers, James Grenning, Bas Vodde
+ * and Arnd R. Strube. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the <organization> nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE EARLIER MENTIONED AUTHORS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL <copyright holder> BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "IEEE754PluginTest_c.h"
+#include <math.h>
+#include <fenv.h>
+
+void set_divisionbyzero_c(void) {
+    float f = 1.0f;
+    f /= 0.0f;
+    (void) f;
+}
+
+void set_overflow_c(void) {
+    float f = 1000.0f;
+    while (f < INFINITY) f *= f;
+}
+
+void set_underflow_c(void) {
+    float f = 0.01f;
+    while (f > 0.0f) f *= f;
+}
+
+void set_invalid_c(void) {
+    float f = (float) sqrt(-1.0f);
+    (void) f;
+}
+
+void set_inexact_c(void) {
+/* feraiseexcept(FE_INEXACT); */ /* Clang ignores -frounding-math */
+    float f = 10.0f;
+    f /= 3.0f;
+    (void) f;
+}
+
```

Comment:
I did. It didn't work simply making the local variables volatile, but making f both global and volatile did work (at least on my computer). Saves me from having to do (void)f as well, so I think it's good now.